vendors/espressif/esp-idf/components/esp32/hwcrypto/sha.c

```diff
@@ -195,7 +195,7 @@ static bool esp_sha_lock_engine_common(esp_sha_type sha_type, TickType_t ticks_t
 
 void esp_sha_unlock_engine(esp_sha_type sha_type)
 {
-    SemaphoreHandle_t *engine_state = sha_get_engine_state(sha_type);
```

Comment:
Should we be changing the ESP-IDF code like this?

Reply:
Return type of the function `sha_get_engine_state` is `SemaphoreHandle_t`. Hence, the variable `engine_state` should be of the type `SemaphoreHandle_t`.